### PR TITLE
[TASK-6] [Test] As a user, I can see the initial surveys list populated

### PR DESCRIPTION
**Pull request by minhnimble**
https://github.com/minhnimble/android-surveys-list/projects/1#card-46082018
 
## What happened

Need to write unit tests for classes used in load surveys list feature.
 
## Insight

**Acceptance Criteria**
- [x] Write unit tests
 
## Proof Of Work
No visible changes, just under the hood tests 😁.

### CODE REVIEW

**Review by manh-t — APPROVED** [review]

File: app/src/test/java/co/nimblehq/ui/screen/main/surveys/SurveysViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            .surveysPagerItemUiModels
+            .test()
+        surveysViewModel.checkAndLoadSurveysListIfNeeded()
+
+        // Assert
+        showLoadingObserver
+            .assertNoErrors()
+            .assertValueCount(2)
+            .assertValueAt(0) { !it }
+            .assertValueAt(1) { !it }
```
Comment (hoangnguyen92dn) [inline app/src/test/java/co/nimblehq/ui/screen/main/surveys/SurveysViewModelTest.kt:72]: I'd suggest to combine with `assertValues`

File: app/src/test/java/co/nimblehq/ui/screen/main/surveys/SurveysViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        surveysPagerItemUiModelsObserver
+            .assertNoErrors()
+            .assertValueCount(1)
+            .assertValue { it.size == sampleSurveysList.size }
+            .assertValue { it == sampleSurveysList.map { survey -> survey.toSurveysPagerItemUiModel() } }
```
Comment (hoangnguyen92dn) [inline app/src/test/java/co/nimblehq/ui/screen/main/surveys/SurveysViewModelTest.kt:78]: I think just check the content of list is enough. Any reason why do we need to always check size first? 🤔

File: app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        ) doReturn Flowable.just(listOf(Survey()))
+
+        // Act
+        val positiveTestSubscriber = useCase.execute(
+            GetSurveysListFlowableUseCase.Input(
+                1,
+                10
+            )
+        )
+            .test()
```
Comment (sleepylee) [inline app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt:44]: this .test() falls from the above line, looks a bit awkward 😅

File: app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        ) doReturn Flowable.error(SurveyError.GetSurveysListError(null))
+
+        // Act
+        val negativeTestSubscriber = useCase.execute(
+            GetSurveysListFlowableUseCase.Input(
+                1,
+                10
+            )
+        )
+            .test()
```
Comment (sleepylee) [inline app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt:66]: same here

**Review by sleepylee — APPROVED** [review]
looking good, just some format comments

File: app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt (hunk 3 above)
Comment (minhnimble) [reply app/src/test/java/co/nimblehq/usecase/survey/GetSurveysListFlowableUseCaseTest.kt:44]: let me reformat it here to better look 😁

**Pull request merged**